snappy/helpers.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/* This module gathers a set of helper functions and structures shared
+   by the various commands in snapweb
+*/
+
+package snappy
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/snapcore/snapd/client"
+)
+
+// IsDeviceManaged determines if the device is in the 'managed' state
+func IsDeviceManaged() bool {
+	client := client.New(nil)
+
+	sysInfo, err := client.SysInfo()
+	if err != nil {
+		return false
+	}
+
+	if sysInfo.OnClassic || sysInfo.Managed {
+		return true
+	}
+
+	return false
+}
+
+func unixDialer(socketPath string) func(string, string) (net.Conn, error) {
+	file, err := os.OpenFile(socketPath, os.O_RDWR, 0666)
+	if err == nil {
+		file.Close()
+	}
+
+	return func(_, _ string) (net.Conn, error) {
+		return net.Dial("unix", socketPath)
+	}
+}
+
+// MakePassthroughHandler maps a snapd API to a snapweb HTTP handler
+func MakePassthroughHandler(socketPath string, prefix string) http.HandlerFunc {
```

Comment:
this looks like a terrible idea. Could you explain what you're trying to achieve?